crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -774,7 +781,22 @@ fn normalize_string(input: &str, quotes: StringQuotes, is_raw: bool) -> Cow<str>
 
     let mut chars = input.char_indices();
 
+    let is_raw = prefix.is_raw_string();
+    let is_fstring = prefix.is_fstring();
+    let mut in_formatted_value = false;
```

Comment:
Does this work with nested formatted values? 

```
f"{expression + f'a{other} + boom '}"
```

Reply:
Good point. I guess we'll need a counter.

On a side note, your example works but the following will fail:
```python
f"{f'{expr}\'foo\''}"
```